content/java/concepts/array-list/terms/indexOf/indexOf.md

```diff
@@ -0,0 +1,85 @@
+---
+Title: '.indexOf()'
+Description: 'Returns the index of the first occurrence of a given element, or -1 if not found.'
+Subjects:
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags:
+  - 'Arrays'
+  - 'Data Types'
+  - 'Elements'
+  - 'Search'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+The **`.indexOf()`** method returns the index of the first occurrence of the specified element in an `ArrayList`. If the element is not found, -1 is returned.
+
+## Syntax
+
+```pseudo
+myArrayList.indexOf(element);
+```
+
+The index of its first occurrence (even if the value if `null`) is returned if the `element` exists. If the `element` can't be found, -1 will be returned.
+
+## Example
+
+The following example features two calls to `.indexOf()` on `animals`. The first will print the index of the first occurrence of the element with the string `"Tiger"`. The second will return -1 because `"Elephant"` does not exist in `animals`:
+
+```java
+// Import the ArrayList class from the java.util package
+import java.util.ArrayList;
+
+public class Main {
+  public static void main(String[] args) {
+    // Create ArrayList of type String named animals
+    ArrayList<String> animals = new ArrayList<String>();
+    animals.add("Lion");
+    animals.add("Tiger");
+    animals.add("Cat");
+    animals.add("Dog");
+    animals.add("Tiger");
+    animals.add("Lion");
+    animals.add("Tiger");
+    System.out.println(animals.indexOf("Tiger"));
+    System.out.println(animals.indexOf("Elephant"));
+
+  }
+}
+```
+
+This will print the following output:
+
+```shell
+1
+-1
+```
+
+The `.indexOf()` method does not work with primitive data types.
+
+## Example 2
+
+```java
+public class Main {
+  public static void main(String[] args) {
+//    array
+    int[] arr = {1, 2, 3, 4, 5};
+    // check if a element is peresent in a array
+    // returns errror as arrays dont have indexOf() method
+    System.out.println(arr.indexOf(3));
+
+  }
+}
+```
+```shell
```

Comment:
Could we add the following text between lines # 75 and # 76?

Since arrays do not have an `.indexOf()` method, the following error will be thrown:

Reply:
ya added

Reply:
@Dusch4593  if we add these lines in there we have to keep  this example 2 one as it is can you look into that? i am confused on whether to remove example 2 or not 

imo its better to remove as now i added another 2 examples on why arraylist.indexof method dont accept int but does integer 

look into that and let me know 

thankyou
https://github.com/Codecademy/docs/pull/1454/files#r1062497759

Reply:
that link refers to line where you said its better to remove example 2